plugin/trino-hive/src/test/java/io/trino/plugin/hive/fs/TestTransactionScopeCachingDirectoryLister.java

```diff
@@ -187,6 +187,14 @@ public RemoteIterator<LocatedFileStatus> list(FileSystem fs, Table table, org.ap
             return throwingRemoteIterator(requireNonNull(fileStatuses.get(path)), throwException);
         }
 
+        @Override
+        public RemoteIterator<LocatedFileStatus> listFilesRecursively(FileSystem fs, Table table, Path path)
```

Comment:
Is this correct when we want to list the whole tree under the specified path?

Reply:
Correct, we want to list all _files_ underneath the specified path (not including the intermediate "directory" entries). This is ultimately the goal of a "recursive listing" when generating splits in Hive, but comes with a caveat- we no longer can identify and skip sub-listing "hidden" directories, so we have to filter out the files that are listed underneath those hidden paths after the fact.